cw_ethereum/lib/ethereum_wallet.dart

```diff
@@ -508,4 +508,8 @@ abstract class EthereumWalletBase
   @override
   String signMessage(String message, {String? address = null}) =>
       bytesToHex(_ethPrivateKey.signPersonalMessageToUint8List(ascii.encode(message)));
+
+  Future<String> fetchEnsAddress(String name) async {
+    return await _client.checkEnsName(name);
+  }
```

Comment:
ENS names can contain addresses for different assets other than Ethereum (example: https://app.ens.domains/vikrant.eth). Move this to `/lib/entities/ens_record.dart`